### ISSUE TITLE
reading/writing to txt files doesn't work?

### ISSUE THREAD

**Issue by OrderFromChaos** [jdf/processing.py#230]
I created a data.txt file in my sketch folder and did this:

f = open('data.txt','w')
f.write('this is a test')
f.close()

Although it recognized the name of the file and that it existed within the directory (I tried arbitrarily changing the name and got errors), nothing wrote to the file, nor was it overwritten.

I tried reading a .txt file using f.read(), with the same efficacy.

Checking on the website, I found the [createWriter() syntax](http://py.processing.org/reference/createWriter.html), but copy-pasting the code from the website did not work. [Here's the output from copy-pasting.](http://i.imgur.com/la574sT.png)

Note that I am somewhat new to Python and may be missing something relatively basic.

I am currently using Processing 3.2.3.

**Comment by jdf** [comment]
Try

```python
f = open(sketchPath('data.txt'), 'w')
```

**Comment by OrderFromChaos** [comment]
Thanks for the prompt response!

Your solution worked, but I had to also place the read/write inside of either the setup() or draw() functions. It doesn't work outside of it. When placed inside of there, open('data.txt','w') also worked.

**Comment by jdf** [comment]
The complete program you posted above worked perfectly for me when I tried it, with the addition of `sketchPath()`.

Please post a small, complete example of a non-working program.

**Comment by OrderFromChaos** [comment]
I'm pretty sure this is a weird emergent property, because when I tested after you posted that originally, it worked just fine. Now the code below doesn't work when I formed a new file to test it again:
```
f = open('data.txt','w')
f.write('hello')
f.close()

def setup():
    pass

def draw():
    pass
```
I'm importing peasycam by using add_library() and then import (in my original code), but after removing that and restarting Processing, there was no difference.

Note that either time it worked while inside of the draw() or setup() functions.
